nodes/utils/index.js

```diff
@@ -54,8 +57,50 @@ function addConnectionCredentials (RED, msg, conn, config) {
     return msg
 }
 
+function getThirdPartyWidgets (directory) {
+    const contribs = {}
+    const packagePath = path.join(directory, 'package.json')
+    if (!fs.existsSync(packagePath)) {
+        return contribs
+    }
+    const packageJson = JSON.parse(fs.readFileSync(packagePath, 'utf8'))
+    const getWidgets = (packageJson) => {
+        if (packageJson?.['node-red-dashboard-2']) {
+            // loop over object of widgets & add to contribs object
+            Object.entries(packageJson['node-red-dashboard-2'].widgets).forEach(([widgetName, widgetConfig]) => {
+                contribs[widgetName] = {
+                    package: packageJson.name,
+                    name: widgetName,
+                    src: widgetConfig.output,
+                    path: path.resolve(path.join(directory)),
+                    component: widgetConfig.component
+                }
+            })
+        }
+    }
+    if (packageJson?.['node-red-dashboard-2']) {
+        // this _is_ a dashboard node! get its widgets.
+        getWidgets(packageJson)
+    } else if (packageJson && packageJson.dependencies) {
+        // get widgets from dependencies of this package
+        Object.entries(packageJson.dependencies)?.filter(([packageName, _packageVersion]) => {
+            return packageName.includes('node-red-dashboard-2-')
+        }).forEach(([packageName, _packageVersion]) => {
+            const modulePath = path.join(directory, 'node_modules', packageName)
```

Comment:
This only handles plugins that were installed as modules from a repo and not modules installed from a local source.
eg:  `npm install /nodered/plugins/mywidgets`

The value of  `_packageVersion`  ends up as a relative file path reference to the installation location instead of a version in the package.json file eg: 
```
....
"node-red-dashboard-2-mywidgets":"file:../../../nodered/plugins/mywidgets"
....
```

Please see https://github.com/FlowFuse/node-red-dashboard/pull/1274 where I make reference to this in the comments.

Reply:
My apologies, i missed that as part of your criteria & I agree it is a valid feature and makes sense to include it.

Note however, as an intermediate/alternative means, you can utelise the `nodesDir` by adding entries to it in the settings file that may actually suit your situation better. e.g.

```
nodesDir: [`/path/to/local/ui-node-1`, `/path/to/local/ui-node-2`]
```

I will raise a follow up issue to look at supporting file:// installs too.

Reply:
@theotherwillembotha please confirm how you install local packages and let me know if this works/does not work for you.

Reply:
@theotherwillembotha can you comment or is this resolved?